### ISSUE TITLE
Expense - Missing "Required" and "Request" for Category & Tag and Merchant respectively

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#33063]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.12-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition:
- User is an employee of a Collect workspace on OD.
- The workspace has set Category and Tag to required.

1. Go to the Collect workspace chat.
2. Click + > Request money > Manual.
3. Enter amount > Next.

## Expected Result:
On confirmation page,
- "Required" is shown on Category and Tag row.
- "Request" is shown in Merchant field by default.

## Actual Result:
On confirmation page,
- "Required" is not shown on Category and Tag row.
- "Request" is not shown in Merchant field by default.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/6a507a0e-9441-41b0-8302-acdc35a6e055

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
Production on previous build 
![image](https://github.com/Expensify/App/assets/93399543/37fd9995-b8e5-415e-bd67-8033282dd8a9)

**Comment by neonbhai** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expense - Missing "Required" and "Request" for Category & Tag and Merchant respectively

### What is the root cause of that problem?
We have not passed the right label prop for category and tag [here]()

### What changes do you think we should make in order to solve the problem?
We should pass the prop if policy requires category and tag:
```js
rightLabel={Boolean(props.policy.requiresCategory) ? translate('common.required') : ''} 

```

```js
rightLabel={Boolean(props.policy.requiresTag) ? translate('common.required') : ''}

```

We may also have a `canUseViolations` check.

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
On confirmation page,

"Required" is shown on Category and Tag row.
"Request" is shown in Merchant field by default.

### What is the root cause of that problem?
1. In IOURequestStepConfirmation, we don't set rigthLabel for Tag and Category. 
2. We don't have logic to prevent the user from requesting money if category/tag is required and the user doesn't select category/tag. 

### What changes do you think we should make in order to solve the problem?
In 2 places
https://github.com/Expensify/App/blob/3a44f772e2c381b1cdb025f845551c5f0985bad5/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L713
https://github.com/Expensify/App/blob/3a44f772e2c381b1cdb025f845551c5f0985bad5/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L728

we should add
```
rightLabel={canUseViolations && Boolean(props.policy.requiresCategory) ? translate('common.required') : ''}

```

```
rightLabel={canUseViolations && Boolean(props.policy.requiresTag) ? translate('common.required') : ''}
```

And then in here
https://github.com/Expensify/App/blob/3a44f772e2c381b1cdb025f845551c5f0985bad5/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js#L504

we should add an error handling if category/tag is required and the user doesn't select category/tag. 

```
if (!iouCategory) {
  setFormError('common.error.invalidCategory');
  return;
}
if (!iouTag) {
  setFormError('common.error.invalidTag');
  return;
}

```

In en.ts and es.ts we also need to add the translation for 'common.error.invalidCategory' and 'common.error.invalidTag'
### Result

https://github.com/Expensify/App/assets/141406735/6f9dc92d-4575-40a6-91e5-43381bc739e2

**Comment by shubham1206agra** [comment]
No proposal seems to have complete solution, i.e., missing `Request` in merchant.

**Comment by DylanDylann** [comment]
@shubham1206agra [Updated proposal](https://github.com/Expensify/App/issues/33063#issuecomment-1857387081)

**Comment by shubham1206agra** [comment]
@DylanDylann I don't think your solution is correct for setting merchant

**Comment by DylanDylann** [comment]
@shubham1206agra is this [expected](https://github.com/Expensify/App/issues/33063#issuecomment-1855935143)

**Comment by shubham1206agra** [comment]
No I mean your solution method by which you are setting a merchant

**Comment by DylanDylann** [comment]
@shubham1206agra I don't see any way to setup Merchant to request on olddot. So I think that the reporter want to display request as default value

**Comment by shubham1206agra** [comment]
No I mean the use of useEffect hook is not ideal here 😅

**Comment by DylanDylann** [comment]
@shubham1206agra [Updated proposal](https://github.com/Expensify/App/issues/33063#issuecomment-1857387081) with another way like we did in the date field

**Comment by shubham1206agra** [comment]
@tgolen I just want to confirm the default merchant. Should we set it to Request or it was removed for some reason?

**Comment by tgolen** [comment]
I am not sure. I have not heard of it being a requested change, but that doesn't mean it wasn't. @dylanexpensify @mountiny @luacmartins might know.

**Comment by luacmartins** [comment]
It seems like we still use `Request` as the default merchant, no? What do you mean by `it was removed`?

**Comment by shubham1206agra** [comment]
> It seems like we still use `Request` as the default merchant, no? What do you mean by `it was removed`?

I mean as a default merchant.

**Comment by DylanDylann** [comment]
@luacmartins In case the user enables "Required" category/tags, should we disable request money if user don't select category/tags ?

**Comment by luacmartins** [comment]
IIRC we don't block the user in that case and instead will show a violation. Is that right @cead22?

**Comment by DylanDylann** [comment]
@cead22 Could you help to confirm [this one](https://github.com/Expensify/App/issues/33063#issuecomment-1865984622) ?

**Comment by cead22** [comment]
- Shouldn't this have been assigned to me and IR?
- Which file are we keeping in the end, App/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js?
- I say we close this and I'll handle this in a separate issues as part of the violations project

To answer the question, Carlos is right, we don't block the user

**Comment by shubham1206agra** [comment]
> * Which file are we keeping in the end, App/src/components/MoneyTemporaryForRefactorRequestConfirmationList.js?

Yes

> * I say we close this and I'll handle this in a separate issues as part of the violations project

@cead22 Since @DylanDylann has nearly completed this issue fix, let us finish this issue up.

**Comment by cead22** [comment]
The issue fix is a 3 line change, and we don't have a PR with tests or anything yet. I still think we should close and let the people working on violations fix this. This isn't really a bug, it's a small oversight coming from the fact that we have 2 files doing the same thing, one of which is temporary due to a refactor.

Ultimately, I'll lean on what @miljakljajic and @tgolen want to do

**Comment by mountiny** [comment]
The change @shubham1206agra is referring to is that for expenses we will require a merchant to be set during creation so the `Request` will be removed from there. In case of Scan request too there should be no way to add the merchant during creation so Request wont be default either.

**Comment by mountiny** [comment]
@shubham1206agra With that said, I think you can omit changes to the `Merchant` field in this PR and only handle the category and tag. 

We are handling the merchant changes in here https://github.com/Expensify/App/pull/32486

**Comment by DylanDylann** [comment]
@shubham1206agra The PR is ready for review. Could you help to take a look?

**Comment by miljakljajic** [comment]
@shubham1206agra make sure you accept the offer so that we can pay promptly when the seven day hold is up!

**Comment by miljakljajic** [comment]
@shubham1206agra @DylanDylann payment issued for you both! Do you have suggestions for additional test rail steps we should add for this issue?

**Comment by shubham1206agra** [comment]
Not really. This is an edge case handling and is behind a beta right now.

**Comment by greg-schroeder** [comment]
Unassigning myself as it seems Milja already handled this
